app/models/vm_or_template.rb

```diff
@@ -1833,6 +1837,18 @@ def parent_resource
     parent
   end
 
+  def available_child_options
+    Rbac.filtered(VmOrTemplate.by_ems(ems_id)
+      .where.not(:id => [id] + children.pluck(:ids) + parents.pluck(:id)))
```

Comment:
sorry to do this - I liked the 3 lines better.
I like `parents` over `parent`

I think parent is a ruby method, you can't use  `pluck`.
(it will be a relationship until it is cached, then it will be an array)